integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -396,6 +399,87 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
     });
   });
 
+  describe("Errors", () => {
+    it("raises compensating writes errors", async function () {
+      const person1Id = new BSON.ObjectId("0000002a9a7969d24bea4cf5");
+      const person2Id = new BSON.ObjectId("0000002a9a7969d24bea4cf6");
+      const dogId = new BSON.ObjectId("0000002a9a7969d24bea4cf7");
+
+      const promise = new Promise<void>((resolve, _) => {
+        (async () => {
+          const errorCallback: ErrorCallback = (_, error) => {
+            expect(error).instanceOf(CompensatingWriteError);
+            expect(error.code).to.equal(231);
+            expect(error.isFatal).to.be.false;
+            expect(error.message).to.contain(
+              "Client attempted a write that is outside of permissions or query filters; it has been reverted",
+            );
+
+            const cpe = error as CompensatingWriteError;
```

Comment:
It's a bit annoying that the types of `expect` doesn't declare a type assertion.

You could do this yourself with to avoid the rename:

```suggestion
            if (!(error instanceof CompensatingWriteError)) {
              throw new Error("Expected a CompensatingWriteError");
            }
```

After which `error` will be `CompensatingWriteError`.